drivers/ina3221/ina3221.c

```diff
@@ -666,21 +642,24 @@ int ina3221_calculate_power_uw(int16_t *in_mv, int32_t *in_ua, uint8_t num,
     return i;
 }
 
-void ina3221_ch_align(ina3221_channel_t ch, const void *in_res, void *out_res,
+void ina3221_ch_align(uint16_t ch, const void *in_res, void *out_res,
                       size_t res_val_size)
 {
     uint8_t *in = (uint8_t *)in_res;
-    uint8_t tmp_out[INA3221_NUM_CH][res_val_size];
-    int j = 0;
+    uint8_t *out = (uint8_t *)out_res;
 
-    for (int i = 0; i < INA3221_NUM_CH; i++) {
+    for (int i = INA3221_NUM_CH - 1; i >= 0; i--) {
         if (ch & (1 << i)) {
-            memcpy(&tmp_out[i][0], in + j * res_val_size, res_val_size);
-            j++;
+            int k = 0;
+            for (int j = 0; j < i; j++) {
+                if (ch & (1 << j)) {
+                    k++;
+                }
+            }
+            memcpy(out + i * res_val_size, in + k * res_val_size, res_val_size);
```

Comment:
Care to add some comments, this is non-obvious to me. Maybe also [`bitarithm_test_and_clear()`](https://api.riot-os.org/bitarithm_8h.html#a90bfe935dcde09afd2918a3768ad2b17) could be useful here.

Reply:
An explanation of what the function does is in `ina3221.h`.
It copies the values from `in_res` to `out_res` but if `in_res` contains the values `in: [ch2, ch3, 0]` because channel 1 is disabled, out will be `out: [0, ch2, ch3]`.
The function should also work if `in_res == out_res`.
The old version used a temporary buffer.
The buffer no longer needed because the function starts to copy the values from `in_res` to `out_res` starting at the end of `in_res`.
`[ch2, ch3, 0]` -> `[0, 0, ch3]`
`[ch2, ch3, 0]` -> `[0, ch2, ch3]`

I was always thinking if it would be better to always align the channels correctly:
E.g.

```
int ina3221_read_bus_mv(const ina3221_t *dev, uint16_t ch, int16_t out_mv[INA3221_NUM_CH], uint16_t *flags);

uint16_t ch = INA3221_CH1 | INA3221_CH3;
int16_t out_mv[INA3221_NUM_CH];
ina3221_read_bus_mv(dev, ch, out_mv, NULL); /* assume that there is space for all channels even if channel 2 is disabled. */

```